### PR TITLE
new package: java 21

### PR DESCRIPTION
**Pull request by rawlingsj**
#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [x] The upstream project actually supports multiple concurrent versions.
- [x] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [x] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

### CODE REVIEW

**Comment by rawlingsj** [comment]
starting off as draft to see if the build works.  Would be good to verify the virtual package and provides.

File: openjdk-21.yaml [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+      rm "${{targets.destdir}}"/$_java_home/lib/src.zip
+
+subpackages:
+  - name: "openjdk-21-dbg"
+    description: "OpenJDK 20 Java Debug Symbols"
+    pipeline:
+      - uses: split/debug
+    dependencies:
+      runtime:
+        - openjdk-20
```
Comment (ajayk) [inline openjdk-21.yaml:115]:
```suggestion
        - openjdk-21
```

File: openjdk-21.yaml [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+  - runs: |
+      _java_home="usr/lib/jvm/java-20-openjdk"
+
+      mkdir -p "${{targets.destdir}}"/$_java_home
+      cp -r build/*-server-release/images/jdk/* "${{targets.destdir}}"/$_java_home
+      rm "${{targets.destdir}}"/$_java_home/lib/src.zip
+
+subpackages:
+  - name: "openjdk-21-dbg"
+    description: "OpenJDK 20 Java Debug Symbols"
```
Comment (ajayk) [inline openjdk-21.yaml:110]:
```suggestion
    description: "OpenJDK 21 Java Debug Symbols"
```

File: openjdk-21.yaml [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    pipeline:
+      - uses: split/debug
+    dependencies:
+      runtime:
+        - openjdk-20
+
+  - name: "openjdk-21-jre"
+    description: "OpenJDK 20 Java Runtime Environment"
```
Comment (ajayk) [inline openjdk-21.yaml:118]:
```suggestion
    description: "OpenJDK 21 Java Runtime Environment"
```

**Review by kaniini — APPROVED** [review]

**Pull request merged**